src/Language/PureScript/TypeChecker/Subsumption.hs

```diff
@@ -50,7 +49,7 @@ subsumes' val ty1 (ForAll ident ty2 sco) =
       sko <- newSkolemConstant
       let sk = skolemize ident sko sco' ty2
       subsumes val ty1 sk
-    Nothing -> throwError . errorMessage $ UnspecifiedSkolemScope
+    Nothing -> error "subsumes: unspecified skolem scope"
```

Comment:
Makes sense to move these to `error`, but perhaps we should make these messages more friendly on the off chance they do appear too. Perhaps we could make a `fatalError` function that wraps errors:

```
An internal error ocurred during compilation: "subsumes: unspecified skolem scope"
Please report this at https://github.com/purescript/purescript/issues
```

Something along those lines anyway? So it's not so much that error itself needs to be friendly, but just there's some kind of context that is readable by non-compiler-developers.